lib/ansible/modules/system/mdadm.py

```diff
@@ -0,0 +1,516 @@
+  returned: success
+  type: string
+  version_added: 2.6
+devices:
+  description: The devices that compose the array.
+  returned: success
+  type: list
+  version_added: 2.6
+level:
+  description: Raid level (raid0, raid1...).
+  returned: success
+  type: string
+  version_added: 2.6
+metadata:
+  description: mdadm metadata version.
+  returned: success
+  type: string
+  version_added: 2.6
+name:
+  description: Array name.
+  returned: success
+  type: string
+  version_added: 2.6
+num_devices:
+  description: Number of devices active in the array.
+  returned: success
+  type: string
+  version_added: 2.6
+state:
+  description: Array state return by mdadm.
+  returned: success
+  type: string
+  version_added: 2.6
+uuid:
+  description: Array UUID.
+  returned: success
+  type: string
+  version_added: 2.6
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import re
+
+
+class MDArray:
+    def __init__(self, name, dev, uuid, metadata, level, devices, num_devices, ansmod):
+        self.name = name
+        self.dev = dev
+        self.uuid = uuid
+        self.metadata = metadata
+        self.level = level
+        self.sparses = 0
+        self.num_devices = num_devices
+        if devices is None:
+            self.devices = []
+        else:
+            self.devices = devices
+        self.active = False
+        # Ansible module: to run command or return error
+        self._ansmod = ansmod
+        self.state = "unknown"
+        self.degraded = False
+
+    def __str__(self):
+        return self.name
+
+    def refresh_state(self):
+        if not self.active:
+            self.assemble()
+        real_dev_name = os.path.basename(self.get_real_dev())
+        # Check degraded only if supported (not with raid0)
+        degraded_status_file = "/sys/block/%s/md/degraded" % real_dev_name
+        if os.path.exists(degraded_status_file):
+            with open(degraded_status_file, 'r') as f:
+                if '1' in f.read().rstrip("\n"):
+                    self.degraded = True
+                else:
+                    self.degraded = False
+
+        with open("/sys/block/%s/md/array_state" % real_dev_name, 'r') as f:
+            self.state = f.read().rstrip("\n")
+
+    def get_real_dev(self):
+        # /dev/md/name => /dev/mdXXX
+        if os.path.islink(self.dev):
+            return os.path.realpath(self.dev)
+        return self.dev
+
+    def is_readonly(self):
+        real_dev_name = os.path.basename(self.get_real_dev())
+        with open("/sys/block/%s/md/array_state" % real_dev_name, 'r') as f:
+            if 'readonly' in f.read().rstrip("\n"):
+                return True
+        return False
+
+    def disable_readonly(self):
+        mdadm_cmd = self._ansmod.get_bin_path("mdadm", required=True)
+        rc, stdout, err = self._ansmod.run_command(
+            "%s --readwrite %s" % (mdadm_cmd, self.dev)
+        )
+        if rc > 0:
```

Comment:
Why don't you use _check_rc_ of run_command?


        :kw check_rc: Whether to call fail_json in case of non zero RC.
            Default False

Reply:
I didn't know check_rc
I will update this PR with all comment